### ISSUE TITLE
Reflected Cross-Site Scripting (XSS) in Upload Error Messages

### ISSUE THREAD

**Issue by l4rm4nd** [ciur/papermerge#338]
**Affected versions**: potentially >= 1.5.2

Hey @ciur,

unfortunately, I've again identified an XSS issue. However, the likelihood for exploitation is very low, since it is a reflected self-xss, which cannot be exploited by a remote attacker. 

**Description**

Although the application successfully blocks uploaded filenames with unsafe characters, the displayed error message does not properly sanitize the reflected filename. Therefore, if a user uploads a filename with an XSS payload, the file gets successfully blocked and is not uploaded into the application, but the XSS filename is reflected in the displayed error message and instantly executed by the browser.

Example filenames to trigger XSS:

````
Sun'><img src=x onerror=alert(1)>set.jpg
````
````
XSS"><iframe src=" javascript:alert('XSS');">.pdf
````
**Note**: Filenames with special chars are usually only allowed in Unix/Linux operating systems.

![image](https://user-images.githubusercontent.com/21357789/110308998-19235280-8001-11eb-8579-0186814f74ec.png)

![image](https://user-images.githubusercontent.com/21357789/110309111-3821e480-8001-11eb-95e0-20d8c073c418.png)

**Recommendation**

Escape all untrusted user input before storing it into the database or reflecting it in the web application. If user input is always properly escaped, there might be even no need for blocking specific characters.

**Comment by ciur** [comment]
@l4rm4nd, thanks for detailed report, I highly appreciate it!

**Comment by ciur** [comment]
Here is the [fix](https://github.com/ciur/papermerge-js/commit/88adef6d4d102fcd31c8a270c87a3bba563a1a7a).

**Comment by ciur** [comment]
Fix is now part of [version 2.0](https://github.com/ciur/papermerge/releases/tag/v2.0.0)
